src/dune/opam_create.ml

```diff
@@ -116,10 +116,19 @@ let insert_dune_dep depends dune_version =
 
 let opam_fields project (package : Package.t) =
   let dune_version = Dune_project.dune_version project in
+  let dune_in_deps = List.exists ~f:(function dep -> Package.(Name.equal dep.Dependency.name dune_name)) package.depends in
   let package =
     if dune_version < (1, 11) || Package.Name.equal package.name dune_name then
+      let () =
+        if not (Package.Name.equal package.name dune_name) && not dune_in_deps then
+          User_warning.emit ?is_error:None ?loc:None ?hints:None [
```

Comment:
The location should point to the `depends` field in the `dune-project` file. There's no need to pass None for optional arguments. This applies to the code below.

Formatting can be fixed by running `$ make fmt`.